File: OldSchoolJava/app/src/main/java/com/capco/oldschooljava/FirstFragment.java
package com.capco.oldschooljava;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.capco.oldschooljava.databinding.FragmentFirstBinding;

public class FirstFragment extends Fragment {

    private FragmentFirstBinding binding;
    private FragmentFirstBinding getBinding() {
        if (binding == null) {
            throw new NullPointerException("binding requested while null");
        }
        return binding;
    }

    // make getters to simplify the rest of working with the ui components of this fragment

    private Button getButtonFirst() { return getBinding().buttonFirst; }
    private Button getCalculateButton() { return getBinding().calculateButton; }
    private EditText getLoanAmountEditText() { return getBinding().loanAmountEditText; }
    private double getLoanAmount() { return Double.parseDouble(getLoanAmountEditText().getText().toString()); }
    private EditText getInterestRateEditText() { return getBinding().interestRateEditText; }
    private double getInterestRate() { return Double.parseDouble(getInterestRateEditText().getText().toString()); }
    private EditText getLoanTermEditText() { return getBinding().loanTermEditText; }
    private int getLoanTerm() { return Integer.parseInt(getLoanTermEditText().getText().toString()); }
    private TextView getResultTextView() { return getBinding().resultTextView; }

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {

        binding = FragmentFirstBinding.inflate(inflater, container, false);
        return binding.getRoot();

    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getButtonFirst().setOnClickListener((clickedView) -> {
            NavHostFragment.findNavController(FirstFragment.this)
                    .navigate(R.id.action_FirstFragment_to_SecondFragment);
        });

        getCalculateButton().setOnClickListener((clickedView) -> {
            onClickCalculate();
        });
    }

    private void onClickCalculate() {
        double monthlyPayment = calculateMonthlyPayment(getLoanAmount(), getInterestRate(), getLoanTerm());
        String resultText = "Monthly Payment: $" + String.format("%.2f", monthlyPayment);
        getResultTextView().setText(resultText);
    }

    private double calculateMonthlyPayment(double loanAmount, double interestRate, int loanTerm) {
        double monthlyInterestRate = (interestRate / 100.0) / 12.0;
        int numberOfPayments = loanTerm * 12;
        double monthlyPayment;
        if (monthlyInterestRate == 0.0) {
            monthlyPayment = loanAmount / numberOfPayments;
        } else {
            double interestRateFactor = Math.pow(1.0 + monthlyInterestRate,numberOfPayments);
            monthlyPayment = (loanAmount * monthlyInterestRate * interestRateFactor) / (interestRateFactor - 1);
        }
        return monthlyPayment;
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

}

File: OldSchoolKotlin/app/src/main/java/com/capco/myapplication/FirstFragment.kt
package com.capco.myapplication

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.navigation.fragment.findNavController
import com.capco.myapplication.databinding.FragmentFirstBinding
import kotlin.math.pow

/**
 * A simple [Fragment] subclass as the default destination in the navigation.
 */
class FirstFragment : Fragment() {

    private var _binding: FragmentFirstBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    // make getters to simplify the rest of working with the ui components of this fragment

    private val buttonFirst : Button get() = binding.buttonFirst
    private val calculateButton : Button get() = binding.calculateButton
    private val loanAmountEditText : EditText get() = binding.loanAmountEditText
    private val loanAmount : Double get() = loanAmountEditText.text.toString().toDoubleOrNull() ?: 0.0
    private val interestRateEditText : EditText get() = binding.loanAmountEditText
    private val interestRate : Double get() = interestRateEditText.text.toString().toDoubleOrNull() ?: 0.0
    private val loanTermEditText : EditText get() = binding.loanTermEditText
    private val loanTerm : Int get() = loanTermEditText.text.toString().toIntOrNull() ?: 0
    private val resultTextView : TextView get() = binding.resultTextView

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        _binding = FragmentFirstBinding.inflate(inflater, container, false)
        return binding.root

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        buttonFirst.setOnClickListener {
            findNavController().navigate(R.id.action_FirstFragment_to_SecondFragment)
        }

        calculateButton.setOnClickListener { onClickCalculate() }
    }

    private fun onClickCalculate() {
        val monthlyPayment = calculateMonthlyPayment(loanAmount, interestRate, loanTerm)
        resultTextView.text =
            "Monthly Payment: $${String.format("%.2f", monthlyPayment)}"
    }

    private fun calculateMonthlyPayment(loanAmount: Double, interestRate: Double, loanTerm: Int): Double {
        val monthlyInterestRate : Double = (interestRate / 100) / 12
        val numberOfPayments : Int = loanTerm * 12

        // Calculate the monthly payment using the formula
        val monthlyPayment : Double = if (monthlyInterestRate == 0.0) {
            loanAmount / numberOfPayments
        } else {
            val interestRateFactor = (1.0 + monthlyInterestRate).pow(numberOfPayments);
            (loanAmount * monthlyInterestRate * interestRateFactor) / (interestRateFactor - 1)
        }

        return monthlyPayment
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}